### ISSUE TITLE
sort_by _eval for joined field does not work

### ISSUE THREAD

**Issue by prakashr** [typesense/typesense#1856]
## Description

I am evaluating typesense for a project and new to it. so kindly let me know if my approach is wrong. I am having issues with sorting, below is a min code with which i could reproduce the issue.

I have 2 collections - one containing a product and stock with the details of availability for each product in different stores. I am joining the product and stock and want to sort the results by the availability is any one of the stores. (products having no availability in store_1 and store_2 should come at the end)

## Steps to reproduce

```sh
# remove collections if existing to clean up 
curl -s -X DELETE "http://localhost:8108/collections/product" -H "x-typesense-api-key: xyz" 
curl -s -X DELETE "http://localhost:8108/collections/stock" -H "x-typesense-api-key: xyz" 

curl -s -X POST "http://localhost:8108/collections" \
     -H "Content-Type: application/json" \
     -H "x-typesense-api-key: xyz" \
     -d '{
           "name": "product",
           "fields": [
             { "name": "entity_id", "type": "string", "facet": false, "optional": false },
             { "name": "name", "type": "string", "facet": false, "optional": false, "sort": true }
           ]
         }' 
	
curl -s -X POST "http://localhost:8108/collections" \
     -H "Content-Type: application/json" \
     -H "x-typesense-api-key: xyz" \
     -d '{
           "name": "stock",
           "default_sorting_field": "entity_id",
           "fields": [
             { "name": "entity_id", "type": "string", "reference": "product.entity_id", "index": true, "sort": true },
             { "name": "store_.*", "type": "bool", "sort": true }
           ]
         }'

curl -s -X POST "http://localhost:8108/collections/product/documents/import" \
     -H "Content-Type: application/json" \
     -H "x-typesense-api-key: xyz" \
     -d ' {"entity_id": "P0",  "name": "Generic brand Tablet"}
           {"entity_id": "P1", "name": "Tablet from samsung"}
           {"entity_id": "P2", "name": "Tablet from apple"}
           {"entity_id": "P3", "name": "Tablet from oppo"}
           {"entity_id": "P4", "name": "Tablet from vivo"}
           {"entity_id": "P5", "name": "Phone from samsung"}
           {"entity_id": "P6", "name": "Tablet from xiaomi"}' 

curl -s -X POST "http://localhost:8108/collections/stock/documents/import" \
     -H "Content-Type: application/json" \
     -H "x-typesense-api-key: xyz" \
     -d '{"entity_id": "P0", "store_1": true, "store_2": true}
           {"entity_id": "P1", "store_1": false, "store_2": false}
           {"entity_id": "P2", "store_1": false, "store_2": true}
           {"entity_id": "P4", "store_1": true, "store_2": true}
           {"entity_id": "P6", "store_1": false, "store_2": false}
           {"entity_id": "P3", "store_1": true, "store_2": false}
           {"entity_id": "P5", "store_1": true, "store_2": true}' 

echo '\nquerying using sort_by: get products sorted by in stock in any store:\n'

curl -s -X POST "http://localhost:8108/multi_search" \
     -H "Content-Type: application/json" \
     -H "x-typesense-api-key: xyz" \
     -d '{
	   "searches": [
	   	{
			"collection": "product",
			"q": "tablet",
			"query_by": "name",
			"filter_by": "$stock(id: *)",
			"include_fields": "name,$stock(store_1, store_2)",
			"sort_by": "$stock(_eval(store_1:true || store_2:true):desc)"
		}
	   ]
   }' | jq -r '.results [].hits [].document | "\(.name)\t\t\(.stock.store_1)\t\(.stock.store_2)" '
```

## Expected Behavior

tablets that are available in either of the stores should be listed first. and ones that are not available in both should be at the end.
```
Tablet from vivo true true
Generic brand Tablet true true
Tablet from oppo true false
Tablet from apple false true
Tablet from xiaomi false false
Tablet from samsung false false
```

## Actual Behavior

the sort_by is not considered at all.

```
Tablet from xiaomi false false
Tablet from oppo true false
Tablet from apple false true
Generic brand Tablet true true
Tablet from vivo true true
Tablet from samsung false false
```
Additional Info: I have also tried
1.  "sort_by": "$stock(store_1:desc, store_2:desc)" it works (as in the expected behavior)- however since there are multiple stores i want to use _eval
2. tried making store_1 and store_2 as int32 - does not work with _eval
3. when i use "sort_by": "_eval($stock(store1:true)):desc" - it crashes the typesense (segmentation fault). this looks like a parsing bug. let me know if you want me to open a separate bug for this.

## Metadata

**Typesense Version**: 0.26.0

**OS**: docker image for 0.26.0  (official one)

**Comment by kishorenc** [comment]
We are able to reproduce this issue and we will have a fixed build shortly.

**Comment by kishorenc** [comment]
@prakashr 

This is fixed in `27.0.rc29`.

**Comment by prakashr** [comment]
verified that it works on v27 rc. thanks for the quick work @kishorenc @happy-san
